src/main/java/bio/terra/stairway/FlightState.java

```diff
@@ -60,12 +60,12 @@ public void setCompleted(Instant completed) {
   }
 
   public Optional<FlightMap> getResultMap() {
-    return Optional.ofNullable(resultMap);
+    return resultMap;
   }
 
-  public void setResultMap(FlightMap resultMap) {
-    if (resultMap != null) {
-      resultMap.makeImmutable();
+  public void setResultMap(Optional<FlightMap> resultMap) {
```

Comment:
nit: if we keep the no-op empty Optional set on `FlightContext`, let's add a comment here that this always overrides the result map. Otherwise, this doesn't need a comment.

Reply:
Reverted to existing code, null checking moved out of FlightState.